### DISCUSSION TITLE
VPNSecure setup doesn't work - 'user is empty'

### DISCUSSION THREAD

**Discussion by olotob** [qdm12/gluetun#1142] [Q&A] [answered]
Hello.

I am trying to setup with VPNSecure. I am following:

[VPN Secure](https://github.com/qdm12/gluetun/wiki/VPN-Secure)

There is no variable to pass username, however container doesn't start with this error:

`2022-09-09T20:11:18Z ERROR VPN settings: OpenVPN settings: user is empty
`

Certainly key and cert is passed to the container as described in the guide.

I have other clients (for example Ubuntu - non contenerized) and this VPN provider works just fine, and there is no need to add username (just the cert files and the account password). I am unsure what I am missing. Thanks for looking into this!

**Comment by olotob** [comment]
And here is my current compose file:

```version: "3"
services:
  gluetun:
    container_name: gluetun
    image: qmcgaw/gluetun
    cap_add:
      - NET_ADMIN
    environment:
      - VPN_SERVICE_PROVIDER=vpnsecure
      - OPENVPN_KEY_PASSPHRASE=<password>
      - SERVER_COUNTRIES=Netherlands
      - TZ=America/New_York
    ports:
      - 8888:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
    volumes:
      - /mystack/gluetun:/gluetun
```

And volume mounted:

```/mystack/gluetun$ ll
total 2988
drwxr-xr-x  2 root root    4096 Sep  9 20:10 ./
drwxr-xr-x 10 root root    4096 Sep  9 19:49 ../
-rw-r--r--  1 root root    1083 Sep  9 19:45 client.crt
-rw-r--r--  1 root root    1038 Sep  9 19:46 openvpn_encrypted_key
-rw-r--r--  1 root root 3041705 Sep  9 20:11 servers.json
```

**Comment by qdm12** [comment]
Indeed. I had the changes locally but forgot to put them in #848 sorry about that!
e7bfc0746068cfaa0977ca4089da4e0bfb0869e5 should fix it (pull the latest image), let me know if it works for you!
I'll do a v3.31.1 release with this change if it works.

**Comment by olotob** [comment]
Hello!

This helped - thank you for putting fixed version. Now I am a bit further, but looks like still something's missing:

```2022-09-09T20:17:50-04:00 WARN [openvpn] cannot stat file '/etc/openvpn/auth.conf': No such file or directory (errno=2)
2022-09-09T20:17:50-04:00 ERROR [openvpn] --auth-user-pass fails with '/etc/openvpn/auth.conf': No such file or directory (errno=2)
2022-09-09T20:17:50-04:00 ERROR [openvpn] Please correct these errors.
2022-09-09T20:17:50-04:00 INFO [openvpn] Use --help for more information.
2022-09-09T20:17:50-04:00 ERROR [vpn] exit status 1
```

**Comment by qdm12** [comment, marked as answer]
dd7630997b4c7f9fa7b12af42c8587dfbcd59286 should fix this as well as the previous problem.
I tested VPN secure myself and it worked, I really don't know what I've done to lose my code changes somehow, sorry again!
Let me know how it goes 👍

**Reply by olotob** [reply]
It works! I mean, container starts, now I need to explore how to make actual use of it but this issue is solved! Thank you!

**Reply by qdm12** [reply]
Awesome! Released v3.31.1 with the fix 😉
